tests/sentry/tasks/test_commit_context.py

```diff
@@ -335,3 +338,18 @@ def test_multiple_matching_code_mappings_but_only_1_repository_has_the_commit_in
         assert owner.user is None
         assert owner.team is None
         assert owner.context == {"commitId": self.commit_2.id}
+
+    @patch(
+        "sentry.integrations.github.GitHubIntegration.get_commit_context",
+        side_effect=ApiError(text="integration_failed"),
+    )
+    @with_feature("organizations:commit-context-fallback")
+    def test_fallback_if_max_retries_exceeded(self, mock_get_commit_context):
+        with self.tasks() and pytest.raises(MaxRetriesExceededError):
```

Comment:
I can't get this to raise `MaxRetriesExceededError`. @NisanthanNanthakumar any ideas? We rely on celery's auto-retry and I wonder if that's why?